pkg/cmd/miscellaneous.go

```diff
@@ -350,17 +361,63 @@ func getRole() string {
 }
 
 /*
-getTechnicalID returns the Technical Id, which used as clustername of the shoot cluster
+getTargetMapInfo retun garden,project,seed,shoot,shootTechnicalID to global getTargetInfo
 */
-func getTechnicalID() string {
+func getTargetMapInfo() map[string]string {
 	var target Target
 	ReadTarget(pathTarget, &target)
 
-	gardenClientset, err := target.GardenerClient()
-	checkError(err)
-	project, err := gardenClientset.CoreV1beta1().Projects().Get(target.Stack()[1].Name, metav1.GetOptions{})
+	for _, t := range target.Stack() {
+		getTargetInfo[string(t.Kind)] = string(t.Name)
+	}
+
+	shootname := getTargetInfo["shoot"]
+	project, err := getProjectObject()
 	checkError(err)
-	shoot, err := gardenClientset.CoreV1beta1().Shoots(*project.Spec.Namespace).Get(target.Stack()[2].Name, metav1.GetOptions{})
+	shoot, err := getShootObject(*project.Spec.Namespace, shootname)
 	checkError(err)
-	return shoot.Status.TechnicalID
+	getTargetInfo["shootTechnicalID"] = shoot.Status.TechnicalID
+
+	// user for case garden-project-shoot add [seed] = seedName
+	if isTargeted("project", "shoot") {
+		getTargetInfo["seed"] = *shoot.Spec.SeedName
+	}
+
+	return getTargetInfo
+}
+
+/*
+lookup Target Kind ("kind value") return Target Name "name value" from target file ~/.garden/sessions/plantingSession/target
+*/
+func getTargetName(Kind string) string {
+	var target Target
+	ReadTarget(pathTarget, &target)
+	for _, t := range target.Stack() {
+		if string(t.Kind) == Kind {
+			return string(t.Name)
+		}
+	}
+	return fmt.Sprintf("Kind: %s no found", Kind)
```

Comment:
```go
	return fmt.Sprintf("Kind: %s no found", Kind)
```
This is not a proper error handling! Either return an error and handle it from the caller side (or exit the process if you do not expect this case usually)

Here is what happens when you return this string in case the kind is not targeted:

```go
projectName := getTargetName("project") // this will return `Kind: project no found`

 // here you would try to fetch the project with name `Kind: project no found` which will fail of course
p, err := Client.GardenerV1beta1().Projects().Get(projectName, metav1.GetOptions{})
```

Reply:
sure, I was write in my local at the beginning. but remove it later. At that time I was try to find some solution in side the ` getTargetMapInfo` method to handle whether the target name exist in local file before return. then only need call method and return without new object.

fixed.